xbmc/addons/interfaces/GUI/DialogFileBrowser.cpp

```diff
@@ -0,0 +1,334 @@
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogFileBrowser::%s - invalid handler data (additionalShare='%p', strType='%p') on addon '%s'", __FUNCTION__, additionalShare, strType, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = &path;
+
+  VECSOURCES vecShares;
+  if (additionalShare)
+    GetVECShares(vecShares, additionalShare, strPath);
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetSource(strPath, allowNetworkShares, &vecShares, strType);
+  if (bRet)
+    strncpy(&path, strPath.c_str(), maxStringSize);
+  maxStringSize = strPath.length();
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::ShowAndGetImage(void* kodiBase, const char *shares, const char *heading, char &path, unsigned int &maxStringSize)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!shares || !heading)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogFileBrowser::%s - invalid handler data (shares='%p', heading='%p') on addon '%s'", __FUNCTION__, shares, heading, addon->ID().c_str());
+    return false;
+  }
+
+  std::string strPath = &path;
+
+  VECSOURCES vecShares;
+  GetVECShares(vecShares, shares, strPath);
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetImage(vecShares, heading, strPath);
+  if (bRet)
+    strncpy(&path, strPath.c_str(), maxStringSize);
+  maxStringSize = strPath.length();
+  return bRet;
+}
+
+bool Interface_GUIDialogFileBrowser::ShowAndGetImageList(void* kodiBase, const char* shares, const char* heading, char**& path, unsigned int& entries)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return false;
+  }
+
+  if (!shares || !heading)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogFileBrowser::%s - invalid handler data (shares='%p', heading='%p') on addon '%s'", __FUNCTION__, shares, heading, addon->ID().c_str());
+    return false;
+  }
+
+  VECSOURCES vecShares;
+  GetVECShares(vecShares, shares, "");
+
+  std::vector<std::string> pathsInt;
+  bool bRet = CGUIDialogFileBrowser::ShowAndGetImageList(vecShares, heading, pathsInt);
+  if (bRet)
+  {
+    entries = pathsInt.size();
+    path = (char**)malloc(entries*sizeof(char*));
+    for (unsigned int i = 0; i < entries; ++i)
+      path[i] = strdup(pathsInt[i].c_str());
+  }
+  else
+    entries = 0;
+  return bRet;
+}
+
+void Interface_GUIDialogFileBrowser::ClearList(void* kodiBase, char**& path, unsigned int entries)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  if (!addon)
+  {
+    CLog::Log(LOGERROR, "ADDON::Interface_GUIDialogFileBrowser::%s - invalid data", __FUNCTION__);
+    return;
+  }
+
+  if (path)
+  {
+    for (unsigned int i = 0; i < entries; ++i)
+    {
+      if (path[i])
+        free(path[i]);
```

Comment:
Have tried it on other requests and has several times a crash then. Don't know why.